service/controller/clusterapi/v31/resource/tccpazs/create.go

```diff
@@ -159,11 +159,7 @@ func (r *Resource) ensureAZsAreAssignedWithSubnet(ctx context.Context, tccpSubne
 	var azNames []string
 	{
 		for az := range azMapping {
-			// Drop the AZs that are not needed by Cluster or MachineDeployment
-			// CRs anymore. This will allow freeing up resources for later use.
-			if azMapping[az].RequiredByCR {
```

Comment:
Is `RequiredByCR` now still necessary? If not it would be cool to cleanup the dead code.

Reply:
My thinking was to keep it there so that it's easy to resume dynamic AZ management (for whoever works on this in the future).

Currently it cannot cause any harm and if it's still unused in a year and we don't have use for it, It should be rather simple to zap off as well.